receiver/faroreceiver/receiver.go

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package faroreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/faroreceiver"
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	faro "github.com/grafana/faro/pkg/go"
+	farotranslator "github.com/grafana/faro/pkg/translator/faro"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+	"go.uber.org/zap"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
+
+	httpHelper "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/faroreceiver/internal/httphelper"
+)
+
+const faroPath = "/"
+
+func newFaroReceiver(cfg *Config, set *receiver.Settings) (*faroReceiver, error) {
+	set.Logger.Info("FaroReceiver config", zap.Any("config", cfg))
+	err := cfg.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	set.Logger.Debug("Valid FaroReceiver config", zap.Any("config", cfg))
+	r := &faroReceiver{
+		cfg:      cfg,
+		settings: set,
+	}
+
+	transport := "http"
+	if cfg.TLSSetting != nil {
+		transport = "https"
+	}
+
+	r.obsrepHTTP, err = receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{
+		ReceiverID:             set.ID,
+		Transport:              transport,
+		ReceiverCreateSettings: *set,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+type faroReceiver struct {
+	cfg        *Config
+	serverHTTP *http.Server
+
+	nextTraces consumer.Traces
+	nextLogs   consumer.Logs
+
+	obsrepHTTP *receiverhelper.ObsReport
+
+	settings *receiver.Settings
+}
+
+func (r *faroReceiver) Start(ctx context.Context, host component.Host) error {
+	r.settings.Logger.Info("Starting FaroReceiver")
+	if r.nextTraces == nil {
```

Comment:
Are both of these absolutely required? Or could a user find value in just logs or just traces and not want them both to be consumed?

Reply:
So my thinking is as follows, the faro sdk users can send traces/logs, ideally the collector should be setup to receive both, without requiring separate setup. But if you disagree happy to alter.